go/runtime/history/indexer.go

```diff
@@ -45,17 +84,59 @@ func NewBlockIndexer(consensus consensus.Backend, history History) *BlockIndexer
 
 // Start starts the indexer.
 func (bi *BlockIndexer) Start() {
+	bi.mu.Lock()
+	bi.status = StatusStarting
```

Comment:
I would move this to `run`. But since we already have one status there, I would remove `StatusInitializing`. Observe that this can be called multiple times, which will override the status.